src/dune_rules/action_unexpanded.ml

```diff
@@ -415,6 +475,18 @@ let rec expand (t : Dune_lang.Action.t) ~context : Action.t Action_expander.t =
     let+ t = expand t in
     O.Ignore (outputs, t)
   | Progn l ->
+    let rec check_ppx_is_last as_ =
+      match (as_ : Dune_lang.Action.t list) with
+      | Ppx ({ loc; _ }, _) :: _ :: _ ->
+        User_error.raise ~loc
+          [ Pp.text
+              "The (ppx ..) action is currently only supported in the last \
```

Comment:
Is "currently" needed in the error message? Are there any plans to remove that limitation soon?

Reply:
Yes. I'm planning on improving the usability of this feature in future PRs.